bayesian/api_v1.py

```diff
@@ -416,6 +417,47 @@ def post():
         return {'status': 'success'}
 
 
+class UserIntent(ResourceWithSchema):
+    method_decorators = [login_required]
+
+    @staticmethod
+    def get(user, ecosystem):
+        if user is not None and ecosystem is not None:
+            raise HTTPError(404, "Invalid user '{u}' and ecosystem '{e}'.".format(u=user,
+                                                                                  e=ecosystem))
+
+        s3 = StoragePool.get_connected_storage('S3ManualTagging')
+        # get user data
+        return s3.fetch_user_data(user, ecosystem)
```

Comment:
The adapter raises `botocore.exceptions.ClientError` if the given user/ecosystem is not present. It will result in internal server error. It would be nice to capture such exception and return 404 instead of 500.